pandas/core/format.py

```diff
@@ -138,6 +139,7 @@ def __init__(self, series, buf=None, length=True, header=True,
             float_format = get_option("display.float_format")
         self.float_format = float_format
         self.dtype = dtype
+        self.strlen = _strlen_func()
```

Comment:
this should take into account the setting of the option, then everything else will flow thru directly.
make a function somewhere (in formatting) that does something like:

```
get _strlen(...):
    if get_option('unicode.east_asian_len'):
          return east_asian_len
   return strlen
```